apps/server/src/shared/domain/entity/task.entity.ts

```diff
@@ -99,98 +104,147 @@ export class Task extends BaseEntityWithTimestamps implements ILearnroomElement,
+	private getGradedSubmissions(): Submission[] {
+		const submissions = this.getSubmissionItems();
+		const gradedSubmissions = submissions.filter((submission) => submission.isGraded());
 
-		return uniqueSubmittedUserIds;
+		return gradedSubmissions;
 	}
 
-	getNumberOfSubmittedUsers(): number {
-		const submittedUserIds = this.getSubmittedUserIds();
-		const count = submittedUserIds.length;
+	private isSubmittedForUser(user: User): boolean {
+		const submissions = this.getSubmissionItems();
+		const isSubmitted = submissions.some((submission) => submission.isSubmittedForUser(user));
 
-		return count;
+		return isSubmitted;
 	}
 
-	getGradedUserIds(): EntityId[] {
-		const gradedUserIds = this.getSubmissionItems()
-			.filter((submission) => submission.isGraded())
-			.map((submission) => submission.student.id);
-		const uniqueGradedUserIds = [...new Set(gradedUserIds)];
+	private isGradedForUser(user: User): boolean {
+		const submissions = this.getSubmissionItems();
+		const graded = submissions.some((submission) => submission.isGradedForUser(user));
 
-		return uniqueGradedUserIds;
+		return graded;
+	}
+
+	private getListOfSubmittedUserIds(): EntityId[] {
+		let submittedUserIds: EntityId[] = [];
+		const submittedSubmissions = this.getSubmittedSubmissions();
+
+		submittedSubmissions.forEach((submission) => {
+			const memberUserIds = submission.getMemberIds();
+			submittedUserIds = [...submittedUserIds, ...memberUserIds];
+		});
+
+		const uniqueSubmittedUserIds = [...new Set(submittedUserIds)];
+
+		return uniqueSubmittedUserIds;
 	}
 
-	getNumberOfGradedUsers(): number {
-		const gradedUserIds = this.getGradedUserIds();
-		const count = gradedUserIds.length;
+	private getListOfGradeddUserIds(): EntityId[] {
+		let gradedUserIds: EntityId[] = [];
+		const gradedSubmissions = this.getGradedSubmissions();
 
-		return count;
+		gradedSubmissions.forEach((submission) => {
+			const memberUserIds = submission.getMemberIds();
+			gradedUserIds = [...gradedUserIds, ...memberUserIds];
+		});
+
+		const uniqueGradedUserIds = [...new Set(gradedUserIds)];
+
+		return uniqueGradedUserIds;
 	}
 
-	// attention based on this parent use this.getParent() instant
-	getMaxSubmissions(): number {
-		// hack until parents are defined
-		const numberOfStudents = this.course ? this.course.getNumberOfStudents() : 0;
+	private userIsSubstitutionTeacher(user: User): boolean {
+		const isSubstitutionTeacher = this.course ? this.course.userIsSubstitutionTeacher(user) : false;
 
-		return numberOfStudents;
+		return isSubstitutionTeacher;
 	}
 
-	createTeacherStatusForUser(user: User): ITaskStatus {
-		const submitted = this.getNumberOfSubmittedUsers();
-		const graded = this.getNumberOfGradedUsers();
-		const maxSubmissions = this.getMaxSubmissions();
+	public createTeacherStatusForUser(user: User): ITaskStatus {
+		const submitted = this.getListOfSubmittedUserIds().length;
+		const graded = this.getListOfGradeddUserIds().length;
```

Comment:
```suggestion
		const numberOfSubmitters = this.getListOfSubmittedUserIds().length;
		const numberOfSubmittersWithGrade = this.getListOfGradeddUserIds().length;
```
Ich würde es gut finden, wenn wir diese Konstanten gerade nach den Unklarheiten zu diesen Werten ganz präzise benennen. Das Gleiche gilt für den Status für Students.